src/transformers/models/mask_rcnn/modeling_maskrcnn.py

```diff
@@ -0,0 +1,3818 @@
+    rois: torch.FloatTensor = None
+    proposals: torch.FloatTensor = None
+    fpn_hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    attentions: Optional[Tuple[torch.FloatTensor]] = None
+
+
+def reduce_loss(loss, reduction):
+    """Reduce loss as specified.
+
+    Args:
+        loss (`torch.Tensor`):
+            Elementwise loss tensor.
+        reduction (`str`):
+            Options are `"none"`, `"mean"` and `"sum"`.
+
+    Return:
+        `torch.Tensor`: Reduced loss tensor.
+    """
+    reduction_enum = nn.functional._Reduction.get_enum(reduction)
+    # none: 0, elementwise_mean:1, sum: 2
+    if reduction_enum == 0:
+        return loss
+    elif reduction_enum == 1:
+        return loss.mean()
+    elif reduction_enum == 2:
+        return loss.sum()
+
+
+def weight_reduce_loss(loss, weight=None, reduction="mean", avg_factor=None):
+    """Apply element-wise weight and reduce loss.
+
+    Args:
+        loss (`torch.Tensor`):
+            Element-wise loss.
+        weight (`torch.Tensor`, *optional*):
+            Element-wise weights.
+        reduction (`str`, *optional*, defaults to `"mean"`):
+            Same as built-in losses of PyTorch.
+        avg_factor (`float`, *optional*):
+            Average factor when computing the mean of losses.
+
+    Returns:
+        `torch.Tensor`: Processed loss values.
+    """
+    # if weight is specified, apply element-wise weight
+    if weight is not None:
+        loss = loss * weight
+
+    # if avg_factor is not specified, just reduce the loss
+    if avg_factor is None:
+        loss = reduce_loss(loss, reduction)
+    else:
+        # if reduction is mean, then average the loss by avg_factor
+        if reduction == "mean":
+            # Avoid causing ZeroDivisionError when avg_factor is 0.0,
+            # i.e., all labels of an image belong to ignore index.
+            eps = torch.finfo(torch.float32).eps
+            loss = loss.sum() / (avg_factor + eps)
+        # if reduction is 'none', then do nothing, otherwise raise an error
+        elif reduction != "none":
+            raise ValueError('avg_factor can not be used with reduction="sum"')
+    return loss
+
+
+def accuracy(pred, target, topk=1, thresh=None):
+    """Calculate accuracy according to the prediction and target.
+
+    Args:
+        pred (`torch.Tensor`):
+            The model prediction, shape (N, num_class).
+        target (`torch.Tensor`):
+            The target of each prediction, shape (N, ).
+        topk (`int` or `Tuple[int]`, *optional*, defaults to 1):
+            If the predictions in `topk` matches the target, the predictions will be regarded as correct ones.
+        thresh (`float`, *optional*):
+            If not None, predictions with scores under this threshold are considered incorrect.
+
+    Returns:
+        `float` or `Tuple[float]`: If the input `topk` is a single integer,
```

Comment:
it's a list or float that's returned

Reply:
The function can also return a single float.